app/controllers/concerns/checkout_callbacks.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module CheckoutCallbacks
+  extend ActiveSupport::Concern
```

Comment:
I'm wondering why this should be extracted to a `Concern`. Should/will be this `concern` reused somewhere?

Reply:
I doubt it will be reused, unless we move the checkout to the API at some point. In this case it was more to tidy up an already bloated controller which will undoubtedly get larger. All of these little methods are fairly insignificant in terms of the important and complex order-processing work this controller needs to do. This basically cuts the controller size in half, and everything that's shifted to the concern is not really critical, which makes it a lot easier to look through the important stuff without getting lost in a mess of minor details.

It comes down to personal taste I guess, but in this specific case I think it's a useful DX improvement :man_shrugging:

Reply:
I personally don't think it's a good idea to extract to a `Concern` here - there's not a clear case to reuse this concern anywhere, so it's essentially just putting the code in two separate files, right? How are we deciding what goes in which file - what makes some code "important" or "critical" and other code just minor details? It looks like we're effectively just moving anything callback-related to the concern, but if we do that, there's quite a few other controllers that have similarly complex callbacks and I don't think we should have a `Concern` for each one of those.